[CI] EsqlSpecForceStoredLoadingIT test {csv-spec:stats_sparkline.sparklineWithWeightedAvg} failing

**Build Scans:**
- [elasticsearch-intake #42121 / part3](https://gradle-enterprise.elastic.co/s/icc6u3c3dhk4c)
- [elasticsearch-pull-request #156748 / part-3](https://gradle-enterprise.elastic.co/s/qo5bzydwq2ff4)
- [elasticsearch-pull-request #156748 / part-3](https://gradle-enterprise.elastic.co/s/vdktaswcqcmoe)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT" -Dtests.method="test {csv-spec:stats_sparkline.sparklineWithWeightedAvg}" -Dtests.seed=82256EC655DB9205 -Dtests.locale=su-Latn-ID -Dtests.timezone=Asia/Manila -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bcsv-spec%3Astats_sparkline.sparklineWithWeightedAvg%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Data mismatch:
row 0 column 0:a list containing
row 0 column 0: 0: <0.0>
row 0 column 0: 1: <45530.883116883124>
row 0 column 0: 2: <0.0>
row 0 column 0: 3: <0.0>
row 0 column 0: 4: <44817.0>
row 0 column 0: 5: <0.0>
row 0 column 0: 6: <62405.0>
row 0 column 0: 7: <0.0>
row 0 column 0: 8: <49095.0>
row 0 column 0: 9: <51403.83557951482>
row 0 column 0:10: expected <54760.79102564103> but was <54760.791025641025>
row 0 column 0:11: <0.0>
Actual:
sparkline:double
[0.0, 45530.883116883124, 0.0, 0.0, 44817.0, 0.0, 62405.0, 0.0, 49095.0, 51403.83557951482, 54760.791025641025, 0.0]

Expected:
sparkline:double
[0.0, 45530.883116883124, 0.0, 0.0, 44817.0, 0.0, 62405.0, 0.0, 49095.0, 51403.83557951482, 54760.79102564103, 0.0]

```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {csv-spec:stats_sparkline.sparklineWithWeightedAvg}
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test test {csv-spec:stats_sparkline.sparklineWithWeightedAvg} (0.7% fail rate in 458 executions)
- [main] 2 failures in step part-3 (0.8% fail rate in 245 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/EsqlSpecForceStoredLoadingIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/EsqlSpecForceStoredLoadingIT.java
@@ -20,9 +20,12 @@ import java.util.List;
 /**
  * Runs the {@code csv-spec} tests while <strong>requesting</strong> all values to
  * be loaded from {@code stored} fields. This should mostly not change the results.
- * BUT it changes the order of multivalued fields, so:
+ * BUT it changes the order of multivalued fields, and the order in which values
+ * are fed into floating-point aggregations (leading to slightly different results),
+ * so:
  * <ul>
  *     <li>We ignore the order of multivalued fields in the results.</li>
+ *     <li>We allow some tolerance for floating-point results.</li>
  *     <li>
  *         We skip a few tests that have no chance of working with the changed order.
  *     </li>
@@ -69,6 +72,11 @@ public class EsqlSpecForceStoredLoadingIT extends EsqlSpecIT {
 
     @Override
     protected boolean ignoreValueOrder() {
+        return true;
+    }
+
+    @Override
+    protected boolean enableRoundingDoubleValuesOnAsserting() {
         return true;
     }
 }

